controllers/admin/AdminCarriersController.php

```diff
@@ -642,13 +642,13 @@ public function changeZones($id)
      */
     public function getList($id_lang, $order_by = null, $order_way = null, $start = 0, $limit = null, $id_lang_shop = false)
     {
-        parent::getList($id_lang, $order_by, $order_way, $start, $limit, $id_lang_shop);
+        // Replace "0" by the name of the shop directly in SQL query (allowing sort without errors)
+        $this->_select .= sprintf(
+            ', IF(name = "0", "%s", name) AS name',
```

Comment:
why the comma before the IF ?
This method is public and _select is empty by default

Reply:
Because it's not empty as @eternoendless said

https://github.com/PrestaShop/PrestaShop/blob/30f668874fdc21d55eecb5a8fda6a48af317fe28/controllers/admin/AdminCarriersController.php#L138

Reply:
But this line is also in a public method, not inside the constructor :thinking:

Reply:
👍🏾  you can call getList without renderList

Reply:
FYI (and after some searchs) :
* `renderList` is called and implement `$this->_select`
* After that, it calls `parent::renderList` (`classes/controller/AdminController.php`)
* This one calls `$this->getList($this->context->language->id);`
* So it's not empty

Reply:
But I have improved the resiliency.